x-pack/plugins/secops/public/components/page/network/ip_overview/field_renderers.tsx

```diff
@@ -0,0 +1,138 @@
+              <DefaultDraggable
+                id={`${IpOverviewId}-${fieldName}`}
+                field={fieldName}
+                value={locationValue}
+              />
+            </EuiFlexItem>
+          </React.Fragment>
+        );
+      })}
+    </EuiFlexGroup>
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const dateRenderer = (fieldName: string, data: Overview): React.ReactElement => {
+  const value = getOr(null, fieldName, data);
+  return value && isString(value) ? (
+    <PreferenceFormattedDate value={new Date(value)} />
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const autonomousSystemRenderer = (
+  as: AutonomousSystem,
+  flowType: IpOverviewType
+): React.ReactElement => {
+  return as && as.as_org && as.asn ? (
+    <EuiFlexGroup alignItems="center" gutterSize="none">
+      <EuiFlexItem grow={false}>
+        <DefaultDraggable
+          id={`${IpOverviewId}-${flowType}.autonomous_system.as_org`}
+          field={`${flowType}.autonomous_system.as_org`}
+          value={as.as_org}
+        />{' '}
+        /
+        <DefaultDraggable
+          id={`${IpOverviewId}-${flowType}.autonomous_system.asn`}
+          field={`${flowType}.autonomous_system.asn`}
+          value={as.asn}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const hostIdRenderer = (host: HostEcsFields, ipFilter: string): React.ReactElement => {
+  return host.id && host.ip && host.ip.includes(ipFilter) ? (
+    <EuiFlexGroup alignItems="center" gutterSize="none">
+      <EuiFlexItem grow={false}>
+        <DefaultDraggable id={`${IpOverviewId}-host-id`} field={'host.id'} value={host.id}>
+          <HostDetailsLink hostId={host.id} />
+        </DefaultDraggable>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const hostNameRenderer = (host: HostEcsFields, ipFilter: string): React.ReactElement => {
+  return host.id && host.ip && host.ip.includes(ipFilter) ? (
+    <EuiFlexGroup alignItems="center" gutterSize="none">
+      <EuiFlexItem grow={false}>
+        <DefaultDraggable id={`${IpOverviewId}-host-name`} field={'host.name'} value={host.id}>
+          <HostDetailsLink hostId={host.id}>
+            {host.name ? host.name : getEmptyTagValue()}
+          </HostDetailsLink>
+        </DefaultDraggable>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const whoisRenderer = (ip: string) => {
+  return (
+    <>
+      <WhoIsLink domain={ip}>{i18n.VIEW_WHOIS}</WhoIsLink>
+      <ExternalLinkIcon />
+    </>
+  );
+};
+
+export const reputationRenderer = (ip: string): React.ReactElement => {
+  return (
+    <>
+      <VirusTotalLink link={ip}>{i18n.VIEW_VIRUS_TOTAL}</VirusTotalLink>
+      <ExternalLinkIcon />
+      <br />
+      <ReputationLink domain={ip}>{i18n.VIEW_TALOS_INTELLIGENCE}</ReputationLink>
+      <ExternalLinkIcon />
+    </>
+  );
+};
```

Comment:
All of this is very readable :+1: for the small pure functions.

Reply:
Thanks! That was my intent with the start of these renderers. Small, succinct, easy-reading/testing pure functions to quickly display data on the page in a consistent format across the app. I think they end up helping the readability of the `IP Overview` component as a whole as well.